src/server/editorServices.ts

```diff
@@ -2795,11 +2805,13 @@ namespace ts.server {
             const reloadChildProject = (child: ConfiguredProject) => {
                 if (!updatedProjects.has(child.canonicalConfigFilePath)) {
                     updatedProjects.set(child.canonicalConfigFilePath, true);
-                    this.reloadConfiguredProject(child, reason);
+                    this.reloadConfiguredProject(child, reason, /*isInitialLoad*/ false);
                 }
             };
             // try to reload config file for all open files
             openFiles.forEach((openFileValue, path) => {
+                // Invalidate default config file name for open file
+                this.configFileForOpenFiles.delete(path);

```

Comment:
Yes, we dont clear whole map, we only clear the configFile for open script info which the project needs to be reloaded. (its stored in configFileExistenceCache that determines which open files will affect because of change/creation or deletion of config file)